HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,736 @@
+    * automatic associations.
+    *
+    * This value may be changed at any time with a CryptoUpdateTransactionBody.
+    */
+    int32 max_automatic_token_associations = 14;
+}
+```
+
+#### Account Update
+
+Note: the only change to this API is in the specification, which now allows for -1.
+
+```protobuf
+message CryptoUpdateTransactionBody {
+
+    ...
+
+    /**
+    * The maximum number of tokens that an Account can be implicitly associated
+    * with. Any non-negative number indicates the maximum number
+    * of such automatic associations. -1 indicates an unlimited number of
+    * automatic associations.
+    */
+    google.protobuf.Int32Value max_automatic_token_associations = 15;
+}
+```
+
+#### Airdrop
+
+```protobuf
+/**
+* A token airdrop transaction's change on state is dependent on the recieving accounts
+* association with a token or the presence of available automatics associations.
+* Airdrops that result in transfers will see the transfers populated in the record token_transfer_list.
+* Airdrops that consume an association slot will be represented in the record automatioc_association field.
+* Airdrops that result in pending transfers will see the intended transfer captures in the record pending_airdrops field.
+*/ 
+message TokenAirdropTransactionBody {
+    /**
+    * The desired list of tokens to airdrop. If any custom fees are assessed,
+    * the ledger will try to deduct them from the payer of this airdrop,
+    * resolving the transaction to INSUFFICIENT_PAYER_BALANCE_FOR_CUSTOM_FEE if
+    * this is not possible.
+    * The sender must have all the succifient balance at the time of the transaction to support this transaction.
+    *
+    * If the receiver account has already associated with the given tokens, then
+    * the fee to transfer will be identical to a token transfer. If the receiver
+    * has open auto-association slots but has not yet associated with the token,
+    * then the payer will pay for the transfer, association, one renewal credit,
+    * and airdrop fee. If the receiver has not associated with the token and has
+    * no open association slot, then the transfer will not deliver the token,
+    * but will create a `pending airdrop` instead, with the receiver account
+    * listed on the `pending airdrop`. The payer will pay for the transfer,
+    * association, one renewal credit, and airdrop fee.
+    * All token_transfers must succeed (result in transfer or pending state) for the transaction to succeed.
+    *
+    * A max list of up to 10 tokens maybe be proposed for airdrop.
+    */
+    repeated TokenTransferList token_transfers = 1;
+}
+```
+
+TokenAirdrop transaction will support the transfers of value in a similar manner to CryptoTransfer with the difference of only supporting tokens (no HBAR transfer support) and supporting the resulting pending transfer state where a CryptoTransfer would have failed due to the lack of association options.
```

Comment:
Won't `CryptoTransfer` also support producing of a pending transfer state if the receiver is not associated to a token being transferred? It seems that in this case `CryptoTransfer` would behave exactly like a `TokenAirdrop` and would have a higher fee as usual. If I'm right, probably we can further describe this possible path.

Reply:
No, cryptoTransfer will not result in pending.
Only change to `CryptoTransfer` is the sender paying.
Pending transfers are a result of the `TokenAirdrop` transaction only